libsolidity/parsing/ParserBase.cpp

```diff
@@ -44,14 +44,32 @@ int ParserBase::endPosition() const
 
 void ParserBase::expectToken(Token::Value _value)
 {
-	if (m_scanner->currentToken() != _value)
-		fatalParserError(
-			string("Expected token ") +
-			string(Token::name(_value)) +
-			string(" got '") +
-			string(Token::name(m_scanner->currentToken())) +
-			string("'")
-		);
+	Token::Value tok = m_scanner->currentToken();
+	if (tok != _value)
+	{
+		if (Token::isElementaryTypeName(tok)) //for the sake of accuracy in reporting
+		{
+			unsigned firstSize;
+			unsigned secondSize;
+			tie(firstSize, secondSize) = m_scanner->currentTokenInfo();
+			ElementaryTypeNameToken elemTypeName(tok, firstSize, secondSize);
```

Comment:
Can do on the helper function. Just wasn't certain whether I should go about it in the scanner or not. As for the else case...well, not really...because not all of them are elementary type name tokens....so I believe it wouldn't work in this instance if my memory serves me correctly.